services/slam/server_test.go

```diff
@@ -107,6 +129,48 @@ func TestServer(t *testing.T) {
 		test.That(t, extraOptions, test.ShouldResemble, map[string]interface{}{})
 	})
 
+	t.Run("working get pointcloud map stream function", func(t *testing.T) {
+		cloudPath := artifact.MustPath("slam/mock_lidar/0.pcd")
+		pcd, err := os.ReadFile(cloudPath)
+		test.That(t, err, test.ShouldBeNil)
+
+		injectSvc.GetPointCloudMapStreamFunc = func(ctx context.Context, name string) (func() ([]byte, error), error) {
+			reader := bytes.NewReader(pcd)
+			serverBuffer := make([]byte, chunkSizeServer)
+			f := func() ([]byte, error) {
+				n, err := reader.Read(serverBuffer)
+				if err != nil {
+					return nil, err
+				}
+
+				if _, err = reader.Seek(0, io.SeekCurrent); err != nil {
+					return nil, err
+				}
+				return serverBuffer[:n], err
+			}
+
+			return f, nil
+		}
+
+		reqPointCloudMapStream := &pb.GetPointCloudMapStreamRequest{Name: testSlamServiceName}
+		mockServer := makeStreamServerMock()
```

Comment:
Can add in a reset function for the mockServer to allow the same one to be used in multiple tests if desired